src/main/java/com/google/sps/data/Listing.java

```diff
@@ -0,0 +1,61 @@
+package com.google.sps.data;
+import java.util.List;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+
+public final class Listing {
```

Comment:
We normally use AutoValue for data class like this, it saves your effort to implement hash() and equal()